src/main/java/org/jenkinsci/plugins/codesonar/services/AnalysisService.java

```diff
@@ -83,50 +97,91 @@ public Analysis getAnalysisFromUrl(String analysisUrl) throws CodeSonarPluginExc
+        CodeSonarChartConfigData chartConfigData = new CodeSonarChartConfigData(CHART_KIND_BAR, "Number of Warnings", "", "");
+        chartConfigData.getSearch_axes().add(new CodeSonarChartSearchAxis("Warnings", null, String.format("aid:%s",analysisId), ""));
+        chartConfigData.getGroups().add(new CodeSonarChartGroup("Analysis","ID"));
+        
+        //Serialize chart configuration data as JSON
+        Gson gsonSerializer = new GsonBuilder().create();
+        String configdata = gsonSerializer.toJson(chartConfigData);
+
+        //Build char_table request URL
+        URIBuilder uriBuilder = new URIBuilder(baseHubUri);
+        uriBuilder.setPath("/chart_table.json");
+        uriBuilder.addParameter("chart", configdata);
+        uriBuilder.addParameter("filter", Utils.formatParameter(filter, strictQueryParameters));
+        
+        URI requestUri = null;
         try {
-            Long.parseLong(parameter);
-        } catch(NumberFormatException e) {
-            if(strictQueryParameters) {
-                //Surround with double quotes only if parameter is not numeric and the hub supports strict query parameters
-                parameter = String.format("\"%s\"", parameter);
-            }
+            requestUri = uriBuilder.build();
+        } catch (URISyntaxException e) {
+            LOGGER.log(Level.WARNING, "Request URI for Char API contains a syntax error. %nException: {0}%nStack Trace: {1}", new Object[] {e.getMessage(), Throwables.getStackTraceAsString(e)});
+            return null;
+        }
+        
+        InputStream jsonContent = httpService.getContentFromUrlAsInputStream(requestUri.toASCIIString());
+
+        Gson gsonDeserializer = new Gson();
+        CodeSonarChartData chartData = null;
+        try {
+            chartData = gsonDeserializer.fromJson(new InputStreamReader(jsonContent, StandardCharsets.UTF_8), CodeSonarChartData.class);
+            LOGGER.log(Level.INFO, "response chartData={0}", chartData.toString());
+        } catch(JsonSyntaxException e) {
+            LOGGER.log(Level.WARNING, "failed to parse JSON response. %nException: {0}%nStack Trace: {1}", new Object[] {e.getMessage(), Throwables.getStackTraceAsString(e)});
+            return null;
+        }
+        
+        if(chartData.getRows() == null) {
+            LOGGER.log(Level.INFO, "chart data is empty.");
+            //Explicitly returning the warning counter set to zero
+            return new CodeSonarAnalysisWarningCount(0);
+        }
+        
+        if(chartData.getRows().size() == 0) {
+            LOGGER.log(Level.INFO, "chart contains no rows.");
+            //Explicitly returning the warning counter set to zero
+            return new CodeSonarAnalysisWarningCount(0);
+        }
+        
+        if(chartData.getRows().size() > 1) {
+            LOGGER.log(Level.INFO, "chart data does not contain a single row ({0,number,integer}).", chartData.getRows().size());
+            return null;
         }
```

Comment:
I don't think this condition should be merely "INFO"; rather it is indicative of a problem.  If the chart returns multiple rows, then I think it is mostly likely because we used the chart API incorrectly and therefore the plugin code is incorrect.  I think it is far less likely (though never impossible) that the chart API returns the wrong rows.  In any case, it is a problem, and it should be visible.